core/api/src/main/java/org/trellisldp/api/BinaryService.java

```diff
@@ -36,6 +43,30 @@
      */
     CompletableFuture<Void> setContent(BinaryMetadata metadata, InputStream stream);
 
+    /**
+     * Set the content for a binary object using a digest algorithm.
+     *
+     * @implSpec The default implementation will compute a digest while processing the {@code InputStream}.
+     *           A {@code null} response value indicates that the digest algorithm is not supported. In the
```

Comment:
I don't like this at all. A unexceptionally-completed `CompletionStage`, to my mind, should _always_ mean that persistence succeeded. Can we go to an exception instead here?

Reply:
I don't see why we couldn't use an exception here.

Reply:
Cool!